### ISSUE TITLE
Save as copy strange behaviour

### ISSUE THREAD

**Issue by hlavamir** [directus/app#2623]
# Bug Report
## Steps to Reproduce

1. Open an item in any collection
2. Press "save as copy"

## Expected Behaviour
Copy is created and saved....

## Actual Behaviour
In around 50% of cases I get a following error:
Duplicate key "#" for collection "collection name"

It seems to be happenning for "older" (smaller index) items. Some items never work, some always work. Even inside the same collection some do and some don't.

## Technical Details
Directus 8.5.5 (the database was previously managed usign Directus 7)
PHP 7.3.8

**Comment by benhaynes** [comment]
Ahhh, intermittent bugs... how I loathe you.

Thanks @hlavamir, will look into it!

**Comment by hlavamir** [comment]
@benhaynes Thanks!
Let me know if you need any further info from me...

**Comment by benhaynes** [comment]
No problem! It doesn't seem like it would be specific to your install/schema, but if you can reproduce on the online demo or send over your schema (if it's not confidential) that would also help.

**Comment by hlavamir** [comment]
Unfortunatelly I cannot share the database with you, but it seems to be a problem with finding the right new index. I went one by one through all items in one collections and wrote down the numbers.

In the collection there are 26 items (with correct indexes 1 to 26), auto increment is currently at 27. 14 items are status deleted, the rest is either published or draft.

All except one failed
index of the item I tried copy -> index in the error message

Failed attempts
 4 ->  9; 5 -> 10; 7 -> 11; 8 -> 12; 9 -> 13; 10 -> 14; 11 -> 15; 12 -> 16; 13 -> 17; 14 -> 18; 15 -> 19

Then the last one worked (ID 22), but instead of creating an expected ID 27, it created ID 47...

Hope this helps.

**Comment by hlavamir** [comment]
![image](https://user-images.githubusercontent.com/30839810/74838198-716ab200-5323-11ea-8d6c-b08477970b18.png)

![image](https://user-images.githubusercontent.com/30839810/74838333-ba226b00-5323-11ea-914b-439599ab5b14.png)

**Comment by rijkvanzanten** [comment]
The app should send over the values as a POST to the API (to create a new item), so the primary key in this case might be set by the API. I'll leave the issue on the app repo for now until we confirm that's the case

**Comment by hlavamir** [comment]
Hi guys! Would you have any update on this? We are planning to feed data into our database this week and the "save and copy" function would be very useful for that...

Thanks a lot.

**Comment by benhaynes** [comment]
@hemratna — can you confirm what RIjk mentioned?

**Comment by hemratna** [comment]
@benhaynes Yes, APP only sends the values, API is relying on autoincrement of MySQL.

@rijkvanzanten This is related to API, Please move this issue to API repo.
